profiler/data_collector.rst

```diff
@@ -73,6 +78,11 @@ The getters are added to give the template access to the collected information.
 
 .. caution::
 
+    If the data that is not directly related to the request or response, you need to make the data accessible to your DataCollector.
+    This can be achieved by injecting the service that holds the information you intend to profile into your DataCollector.
```

Comment:
Good point. Then why not put this into the doc as a note as well? 
I think this is clear once you are familiar with the concepts of Symfony but not for the n00b reading this for the first time